### PR TITLE
Fix optimizers loss

### PR DESCRIPTION
**Pull request by tinasandu**

### CODE REVIEW

File: src/app/ir/model/cosineembeddingloss.ts [hunk 1]
```diff
@@ -23,6 +23,6 @@ export default class CosineEmbeddingLoss {
   }
 
   public initCode(): string {
-    return `CosineEmbeddingLoss(Margin=${this.Margin}, SizeAverage=${this.SizeAverage}, Reduce=${this.Reduce}, Reduction=${this.Reduction})`;
+    return `CosineEmbeddingLoss(margin=${this.Margin}, size_average=${this.SizeAverage}, reduce=${this.Reduce}, reduction=${this.Reduction})`;
```
Comment (williamprofit) [inline src/app/ir/model/cosineembeddingloss.ts]: the reduction should be quoted as it is a string

Comment (williamprofit) [reply src/app/ir/model/cosineembeddingloss.ts]: i.e. `reduction='{this.Reduction}'`

File: src/app/ir/model/embedding.ts [hunk 2]
```diff
@@ -31,6 +31,6 @@ export default class Embedding {
   }
 
   public initCode(): string {
-    return `Embedding(NumEmbeddings=${this.NumEmbeddings}, EmbeddingDim=${this.EmbeddingDim}, PaddingIdx= ${this.PaddingIdx}, MaxNorm= ${this.MaxNorm}, NormType= ${this.NormType}, ScaleGradByFreq= ${this.ScaleGradByFreq}, Sparse= ${this.Sparse}, Weight= ${this.Weight})`;
+    return `Embedding(NumEmbeddings=${this.NumEmbeddings}, EmbeddingDim=${this.EmbeddingDim}, PaddingIdx= ${this.PaddingIdx}, MaxNorm= ${this.MaxNorm}, NormType= ${this.NormType}, ScaleGradByFreq= ${this.ScaleGradByFreq}, Sparse= ${this.Sparse}, weight= ${this.Weight})`;
```
Comment (williamprofit) [inline src/app/ir/model/embedding.ts]: looking at the doc, the args should be snake case https://pytorch.org/docs/stable/generated/torch.nn.Embedding.html

also extra space after = (sorry for nitpicking :c)

Comment (williamprofit) [reply src/app/ir/model/embedding.ts]: this seems to be true for quite a few other files as far as i can tell

Comment (tinasandu) [reply src/app/ir/model/embedding.ts]: I changed all of them just now. Was fixing only loss and optimizers yesterday.

**Review by williamprofit — COMMENTED** [review]
Looking good, two more things as far as I can tell,

the loss funcs should be prefixed with `nn.` as they're part of the nn module afaik

i'm not sure where that comes from but the default value for type `Optional[torch.Tensor]` should be `None` or a pytorch tensor (see https://pytorch.org/docs/stable/generated/torch.nn.NLLLoss.html)

i think easiest is to just set options to None as default value

**Review by williamprofit — APPROVED** [review]
